src/jit/morph.cpp

```diff
@@ -7689,12 +7706,36 @@ GenTreePtr Compiler::fgMorphCall(GenTreeCall* call)
                 //
                 if (call->IsImplicitTailCall())
                 {
-                    if (varDsc->lvHasLdAddrOp || varDsc->lvAddrExposed)
+                    if (varDsc->lvHasLdAddrOp)
                     {
                         hasAddrExposedVars = true;
                         break;
                     }
-                    if (varDsc->lvPromoted && varDsc->lvIsParam)
+                    else if (varDsc->lvAddrExposed)
```

Comment:
Oh, got it now.  Yeah, I was just mechanically splitting the `A || B` without thinking about the implications of the `break`.  Looking at it now, I think it reads better w/o `else`, will update.